umbraco/Umbraco-CMS
# V10/bugfix/10502 listview node permissions fix

### Prerequisites

- [X ] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes https://github.com/umbraco/Umbraco-CMS/issues/10502

### Description

When a node has a list view enabled, the list view actions do not the parent's granular node permissions do not cascade down to the children.   Note that if the user clicks into the node, they do thee the action in the Actions menu. 

Also if you're using a list view with the "Edit in Infinite Editor", this would mean that users cannot do these actions (since they don't get the action menu on the node details in the infinite editor)

To test:

1. In the content tree, create an item with a list view, and  a selection of children.  
2. Create a new user group  setup the permissions. 
    a. Set the main permissions to not allow a permission (delete/publish/unpublish/move/copy).
    b. In granular permissions. choose the list node made in step 1, and allow the given permission for that node (eg  delete/publish/unpublish/move/Copy).
5.  Login as a user with this permission and validate that they can do the action in the list view and on the details but can't do this action on other items.


There also was an issue with the permission check for unpublish. It was using "U" instead of "Z".  It looks like there was an old comment in there related to how the unpublish logic used to work (based on the publish permission), but it looks like this should be updated so i did that too.
